File: CPP/PerfectSquare.cpp
/* CPP program to find if x is a
perfect square.*/
#include <bits/stdc++.h>
using namespace std;
 
bool isPerfectSquare(long double x)
{
    if (x >= 0) {
 
        long long sr = sqrt(x);
         
        return (sr * sr == x);
    }

    return false;
}
 
int main()
{
    long long x = 2502;
    if (isPerfectSquare(x))
        cout << "Yes";
    else
        cout << "No";
    return 0;
}

File: Java/PerfectSquare.java
/* Java program to find if x is a
 perfect square.*/
class PerfectSquare {

	static boolean isPerfectSquare(int x)
	{
		if (x >= 0) {

			int sr = (int)Math.sqrt(x);
			
			return ((sr * sr) == x);
		}
		return false;
	}

	public static void main(String[] args)
	{
		int x = 2502;

		if (isPerfectSquare(x))
			System.out.print("Yes");
		else
			System.out.print("No");
	}
}

